Sources/BraintreePayPal/BTPayPalReturnURL.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+enum BTPayPalAppSwitchReturnURLState {
+    case unknownPath
+    case succeeded
+    case canceled
+}
+
+/// This class interprets URLs received from the PayPal app via app switch returns.
+///
+/// PayPal app switch authorization requests should result in success or user-initiated cancelation. These states are communicated in the url.
+struct BTPayPalReturnURL {
```

Comment:
This file was renamed but is showing additional diff. Adding logic for the web flow in the `init` as well as adding `isValidURLAction` plus `action` methods here are the actual changes.